### PR TITLE
fixing empty description, refs #14

### PR DESCRIPTION
**Pull request by behrisch**

### CODE REVIEW

File: trac-hub [hunk 1]
```diff
@@ -165,14 +165,17 @@ class Migrator
...
       time: ticket[:time],
       author: ticket[:reporter],
       field: "comment",
       oldvalue: nil,
       newvalue: ticket[:description],
-    })["body"]
+    })
+    if desc != nil
+      body += desc["body"]
+    end
```
Comment (mavam) [inline trac-hub]: A one-liner looks less noisy here:

```ruby
body += desc["body"] if desc
```

**Review by mavam — CHANGES_REQUESTED** [review]
Thanks for submitting a patch. I just have a minor style nitpick.

**Comment by behrisch** [comment]
change made

**Comment by mavam** [comment]
Okay, now the final part: have you *tested* an *verified* that it produces the desired result? Looking at the surrounding code, I see:

```cpp
body += "\n\n" if not @singlepost
desc = ticket_change(@singlepost, {
  ticket: ticket[:id],
  time: ticket[:time],
  author: ticket[:reporter],
  field: "comment",
  oldvalue: nil,
  newvalue: ticket[:description],
})
body += desc["body"] if desc
```

Does this make sense in the bigger picture? In particular, does it make sense to add two newlines if the re is no description?

**Comment by behrisch** [comment]
No, my intention was not to produce a nice result (yet), just to get the script running and to be honest I have a hard time doing so. I already hit the limit of the GitHub API (which is quite low with only 60) and labels do not work for me either. Nevermind, but I will stop here.

**Comment by mavam** [comment]
Okay, then let's put this PR on hold. I can't tell whether it's a partial fix at this point or whether it addresses the empty-body problem in a sensible way.

**Comment by coldfix** [comment]
Hi, just noticed this.

This PR addresses an issue I hadn't encountered when doing my migration: null description on the issue text. However, when fixing it, you should also keep intact the message who created the issue, by e.g. adding a `force` parameter to `ticket_change` that prevents from returning `nil`.

I'm a bit confused:

- you hit the limit? The script uses now an API that should not trigger the github abuse detection. Can you tell the exact error?
- what happens with labels?

Best, Thomas

**Comment by behrisch** [comment]
Thanks for having a look.

I did not hit an abuse limit but rather the general limit on anonymous api queries because the initial query is made without authorization AFAICT. I added an authorization header around line 50 and that helped.

The second point was, that the labels were always empty. I think already the mapping import from the config did fail somehow but did not have a further look because I also needed the labels to be generated and from my understanding of the code the script does not do that.

**Comment by coldfix** [comment]
> I did not hit an abuse limit but rather the general limit on anonymous api queries because the initial query is made without authorization AFAICT. I added an authorization header around line 50 and that helped.

Good point, thanks. The script had code to mitigate rate-limit problems. However, I removed it for the migration that I performed in an attempt to simplify overall the structure of the script. Also, after several changes I made, it was doing a lot less github requests and less likely to hit any limit. In fact, I'm a bit surprised that it did hit the limit in your case - the request at line 50 is executed just a single time. However, if adding an authorization header helps to avoid the problem, this is a very simple change that I will provide with another pull-request.

> The second point was, that the labels were always empty. I think already the mapping import from the config did fail somehow but did not have a further look because I also needed the labels to be generated and from my understanding of the code the script does not do that.

The script assumes that labels exist already on github and the config file allows to specify a mapping between labels on Trac and github side. I think, this is generally a good approach because typically you want to have just a few curated labels on github. The other information not captured by labels can easily be inserted in the second line of the issue body anyway.

Sorry to hear that this was not what you needed.

**Pull request closed without merging**